share/functions/realpath.fish

```diff
@@ -6,8 +6,10 @@
 # If we see that there is an external command by that name we just return. That will cause fish to
 # run the external command. On the other hand, if an external command isn't found we define a
 # function that will provide fallback behavior.
-if not type -q -P realpath
-    function realpath --description 'fallback realpath implementation'
-        builtin fish_realpath $argv[-1]
-    end
+if not command -v realpath > /dev/null
+	if command -v grealpath > /dev/null
+		alias realpath=grealpath
+	else
+		alias realpath=fish_realpath
```

Comment:
> I mean - what is it supposed to do? The manpage acts like it is supposed to use a better realpath, like GNUs, if installed. It does on Linux. It does on your system since you indicated you unprefixed the homebrew utilities.

The man page implies nothing of the sort. The fish_realpath(1) man page wording can certainly be improved. Especially with respect to `--canonicalize-existing`. But I don't see how you reached that conclusion. It seems to me that you simply dislike that our fallback implementation doesn't implement capabilities you believe are useful but all the available evidence suggests are so seldom used that our current fallback is "good enough". If you want to expend the effort to provide a more feature full fallback no one is stopping you.